source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -217,6 +244,128 @@ std::string encodeState(const std::string& original_request_url, const std::stri
   return Base64Url::encode(json.data(), json.size());
 }
 
+/**
+ * Encrypt a plaintext string using AES-256-CBC.
+ */
+std::string encrypt(const std::string& plaintext, const std::string& secret,
```

Comment:
The ID and Access tokens can also be encrypted to provide an addtional layer of protection. We can address this in a follow-up PR.